### ISSUE TITLE
API Mapping example

### ISSUE THREAD

**Issue by thangngoc89** [marmelab/ng-admin#272]
I'm following API mapping document to map my api with ng-admin but I'm not success.
My API wrap all data inside data value and send back with meta like this:
![image](https://cloud.githubusercontent.com/assets/3049054/5992192/16605746-aa51-11e4-9f7c-98b8edc008a0.png)

Here is my code
```
RestangularProvider.addElementTransformer('users', function(element) {

            for (var key in element.data) {
                element[key] = element.data[key];
            }
            return element;
        });
```
In console log it shows
```
Error: Response for getList SHOULD be an array and not an object or something else
```

**Comment by Xennis** [comment]
My configuration:

```
RestangularProvider.addResponseInterceptor(function(data, operation, what, url, response) {
	if (operation == "getList") {
		data = data.items;
	}
	return data;
});
````

In my case the data array is in an `items` fields instead of a `data` field, like in your case.

**Comment by thangngoc89** [comment]
@Xennis  thanks for your help. It works now. ng-admin was developed really fast. It made me hard to follow

**Comment by thangngoc89** [comment]
@Xennis  another problem with API mapping is pagination. You can see in my picture above, total count was store in : 
`data.meta.pagination.total`

I edited your given code above : 
```
app.config(function(RestangularProvider) {
        RestangularProvider.addResponseInterceptor(function(data, operation, what, url, response) {
            if (operation == "getList") {
                response.totalCount = data.meta.pagination.total;
                data = data.data;
            }
            console.log(response);
            return data;
        });
        RestangularProvider.addFullRequestInterceptor(function(element, operation, what, url, headers, params, httpConfig) {
            if (operation == 'getList' && what == 'entityName') {
                params.page = params._page;
                params.limit = params._perPage;
                delete params._page;
                delete params._perPage;
            }
            return { params: params };
        });
    });
```

What I get from browser console : 
```
Object {data: Object, status: 200, headers: function, config: Object, statusText: "OK"…}
config: Object
data: Array[30]
headers: function (name) {
status: 200
statusText: "OK"
totalCount: 777
__proto__: Object
```
But there is no pagination at all.

**Comment by Xennis** [comment]
You made an mistake, change `response.totalCount = data.meta.pagination.total;` to `response.totalCount = meta.pagination.total;`

**Comment by Xennis** [comment]
It should work with the line `response.totalCount = data.meta.pagination.total;`, which you added. So I don't know, may check [API-mapping.md](https://github.com/marmelab/ng-admin/blob/master/doc/API-mapping.md).

**Comment by thangngoc89** [comment]
Its really weird. In console log, I can see totalCount was set 
![image](https://cloud.githubusercontent.com/assets/3049054/5993340/0a5bc08c-aa80-11e4-89ec-009ef7b3eaa5.png)

**Comment by fzaninotto** [comment]
Are you using infinite pagination?

**Comment by thangngoc89** [comment]
@fzaninotto  sorry. It's my bad. I'm using ng-admin 0.4.0. Updated to master and everything are fine now
